contracts/BSCValidatorSet.sol

```diff
@@ -838,6 +841,48 @@ contract BSCValidatorSet is IBSCValidatorSet, System, IParamSubscriber, IApplica
     return voteAddrs;
   }
 
+  function setpreviousVoteAddrSet() private {
+    uint n = previousVoteAddrSet.length;
+    uint m = validatorExtraSet.length;
+
+    if (n>m) {
+      for (uint i=m; i<n; ++i) {
+        previousVoteAddrSet.pop();
+      }
+    }
+
+    uint k = n < m ? n:m;
+    for (uint i; i<k; ++i) {
+      if (!BytesLib.equal(previousVoteAddrSet[i], validatorExtraSet[i].voteAddress)) {
+        previousVoteAddrSet[i] = validatorExtraSet[i].voteAddress;
+      }
+    }
+
+    if (m>n) {
+      for (uint i=n; i < m; ++i) {
+        previousVoteAddrSet.push(validatorExtraSet[i].voteAddress);
+      }
+    }
+  }
+
+  function isMonitoredForMalicousVote(bytes calldata voteAddr) external override view returns (bool) {
+    uint m = validatorExtraSet.length;
+    for (uint i; i<m; ++i) {
+      if (BytesLib.equal(voteAddr, validatorExtraSet[i].voteAddress)) {
+        return true;
+      }
+    }
+
```

Comment:
`validatorExtraSet` and `previousVoteAddrSet` may not match, as the order is not the exact same.